From 243884050555dc7025d0acc2b97958848144af06 Mon Sep 17 00:00:00 2001
From: Ken Chalmers <ken.chalmers@amd.com>
Date: Mon, 26 Jun 2017 13:02:30 -0400
Subject: [PATCH 2557/2831] drm/amd/display: RV stereo support

Fix moving directly from frame packed to frame sequential mode: disable
OTG_3D_STRUCTURE_EN if the stereo mode is not frame packed.

Change-Id: I67d1a9b112f438b881104f4fe3ab1cdd41fedbdc
Signed-off-by: Ken Chalmers <ken.chalmers@amd.com>
Reviewed-by: Vitaly Prosyak <Vitaly.Prosyak@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/dcn10/dcn10_timing_generator.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_timing_generator.c b/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_timing_generator.c
index 83efbec..58fb29f 100644
--- a/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_timing_generator.c
+++ b/drivers/gpu/drm/amd/display/dc/dcn10/dcn10_timing_generator.c
@@ -1082,11 +1082,11 @@ static void dcn10_enable_stereo(struct timing_generator *tg,
 			REG_UPDATE(OTG_STEREO_CONTROL,
 				OTG_DISABLE_STEREOSYNC_OUTPUT_FOR_DP, 1);
 
-		if (flags->PROGRAM_STEREO && flags->FRAME_PACKED)
+		if (flags->PROGRAM_STEREO)
 			REG_UPDATE_3(OTG_3D_STRUCTURE_CONTROL,
-				OTG_3D_STRUCTURE_EN, 1,
-				OTG_3D_STRUCTURE_V_UPDATE_MODE, 1,
-				OTG_3D_STRUCTURE_STEREO_SEL_OVR, 1);
+				OTG_3D_STRUCTURE_EN, flags->FRAME_PACKED,
+				OTG_3D_STRUCTURE_V_UPDATE_MODE, flags->FRAME_PACKED,
+				OTG_3D_STRUCTURE_STEREO_SEL_OVR, flags->FRAME_PACKED);
 
 	}
 
-- 
2.7.4

